src/dotnet/AutoRest/JsonRpc/CallerResponse.cs

```diff
@@ -32,10 +32,16 @@ public CallerResponse(string id)
         public bool SetCompleted(JToken result)
         {
             T value;
-            if (typeof(T) == typeof(bool?))
+            Func<object, bool> trueLikeValue = obj => obj != null && !0.Equals(obj) && !false.Equals(obj) && !"".Equals(obj);
```

Comment:
are we copying this code from some other place where this is being done as well?

Reply:
Yes, from [here](https://github.com/Azure/autorest.common/blob/master/src/JsonRpc/CallerResponse.cs#L39). Will add comment

Reply:
ok, do we need to have the code copied there? or is there a way for us to depend on the autorest.common package? (even if we don't do it as part of this PR)

Reply:
We need to copy for now